Competition2017/src/competition/subsystems/autonomous/MoveToHopperCommandGroup.java

```diff
@@ -0,0 +1,52 @@
+package competition.subsystems.autonomous;
+
+import com.google.inject.Provider;
+
+import competition.subsystems.drive.commands.DriveForDistanceCommand;
+import competition.subsystems.drive.commands.RotateToHeadingCommand;
+import competition.subsystems.vision.commands.RotateRobotToBoilerCommand;
+import edu.wpi.first.wpilibj.command.CommandGroup;
+import xbot.common.properties.DoubleProperty;
+import xbot.common.properties.XPropertyManager;
+
+public class MoveToHopperCommandGroup extends CommandGroup{
+    private DoubleProperty angleToParallel;
+    private DoubleProperty distanceFromBoiler;
+    private DoubleProperty angleToBoiler;
+    private DoubleProperty distanceFromTurningPointToBoiler;
+
+    public MoveToHopperCommandGroup(
+            XPropertyManager propMan,
+            Provider<RotateToHeadingCommand> rotateToHeadingProvider,
+            Provider<DriveForDistanceCommand> driveForDistanceProvider,
+            Provider<RotateRobotToBoilerCommand> rotateRobotToBoilerProvider){
+        angleToParallel = propMan.createPersistentProperty("Angle to line parallel to boiler", 3);
+        distanceFromBoiler = propMan.createPersistentProperty("distance from robot to boiler", 6);
+        angleToBoiler = propMan.createPersistentProperty("angle to face boiler", 70);
+        distanceFromTurningPointToBoiler = propMan.createPersistentProperty("distance from turning point to boiler", 10);
+        
+        RotateRobotToBoilerCommand rotateRobotToBoiler1 = rotateRobotToBoilerProvider.get();
+        this.addSequential(rotateRobotToBoiler1);
+
+        RotateToHeadingCommand rotateToParallelCommand = rotateToHeadingProvider.get();
+        rotateToParallelCommand.setTargetHeadingProp(angleToParallel);
+        this.addSequential(rotateToParallelCommand);
+
+        DriveForDistanceCommand driveAlongParallel = driveForDistanceProvider.get();
+        driveAlongParallel.setDeltaDistance(distanceFromBoiler.get() * Math.cos(angleToParallel.get()));
```

Comment:
This value is only known when the command group starts running. Save the `DriveForDistanceCommand` instance to a private variable, and then in `initialize` for the command group you can compute the distance value and set it.